src/watcher.ts

```diff
@@ -321,7 +321,11 @@ module.exports = class MonoplasmaWatcher extends EventEmitter {
                 return block.timestamp * 1000
             })()
         }
-        return await this.blockTimestampCache[blockNumber]
+        if (this.blockTimestampCache[blockNumber].constructor.name === "Promise") {
+            const timestamp = await this.blockTimestampCache[blockNumber]
+            this.blockTimestampCache[blockNumber] = timestamp
+        }
+        return this.blockTimestampCache[blockNumber]
```

Comment:
curious why?

Reply:
so that the function would actually put the number in place of the promise after it resolves instead of leaving a resolved Promise in there. This was so that I could extract the cache easily into a file (see https://github.com/streamr-dev/data-union-server/blob/master/scripts/watch_dataunion.js#L87)